### PR TITLE
feat/js-array-methods/test-for-challenges

### PR DESCRIPTION
**Pull request by PReichetanz**
## Update `map` and `filter` challenges for new format

rendered: https://github.com/neuefische/web-exercises/tree/feat/js-array-methods/challenges-test/sessions/js-array-methods

- add readme files
- add tests

## Codesandboxes
- [map](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-array-methods/challenges-test/sessions/js-array-methods/map?file=/README.md)
- [filter](https://codesandbox.io/s/github/neuefische/web-exercises/tree/feat/js-array-methods/challenges-test/sessions/js-array-methods/filter?file=/README.md)

refers to neuefische/web-curriculum-new-format#130

### CODE REVIEW

**Review by CJAkkuec — COMMENTED** [review]
LGTM 🚀 (but I'm not too familiar with tests -> third pair of eyes?)

File: sessions/js-array-methods/filter/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+This codesandbox offers a some exercises to recap the array method `filter`. Let's dive directly into it!
+
+## Task
+
+In the [index.js](index.js) file, there is a `cards` array with three objects.
+
+Below this array, you will find a couple of variables (e.g. `onlyCardsWithTagHTML`). They are all initialized with `null`.
+
+And now it's your turn: delete the value `null` and write code so that the variables have a value corresponding their name. Use `filter` to achieve this goal.
```
Comment (Kheirah) [inline sessions/js-array-methods/filter/README.md:11]:
```suggestion
And now it's your turn: delete the value `null` and write code such that the variables have a value corresponding to their name. Use `filter` to achieve this goal.
```

File: sessions/js-array-methods/map/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+This codesandbox offers a some exercises to recap the array method `map`. Let's dive directly into it!
+
+## Task
+
+In the [index.js](index.js) file, there is a `cards` array with three objects.
+
+Below this array, you will find a couple of variables (e.g. `lowerCaseAnswers`). They all are initialized with `null`.
+
+And now it's your turn: delete the value `null` and write code so that the variables have a value corresponding their name. Use `map` to achieve this goal.
```
Comment (Kheirah) [inline sessions/js-array-methods/map/README.md:11]:
```suggestion
And now it's your turn: delete the value `null` and write code such that the variables have a value corresponding to their name. Use `map` to achieve this goal.
```

**Review by Kheirah — APPROVED** [review]
Tests are all working fine. I was merely surprised by the map exercise about the constant "questionsAndAnswersTogether". Took me some tinkering with due to the expected quotation marks in between the questions and answers. If it's on purpose to stimulate students to think about how to concatenate the individual parts, then it's a good exercise. Bonus points for being slightly annoying😃

My solutions:
[JS Map](https://codesandbox.io/s/js-array-methods-map-szo1cd?file=/index.js:699-748)
[JS Filter](https://codesandbox.io/s/js-array-methods-filter-sy0kcr)

**Pull request merged**